tests/backup/backup_test.go

```diff
@@ -246,71 +249,189 @@ var _ = Describe("{BasicBackupCreation}", func() {
+				backupName = fmt.Sprintf("%s-%s", BackupNamePrefix, namespace)
+				CreateBackup(backupName, SourceClusterName, backupLocation, backupLocationUID, []string{namespace},
 					labelSelectors, orgID, clusterUid, preRuleNameList[0], preRuleUid, postRuleNameList[0], postRuleUid, ctx)
 			}
 		})
 		Step("Restoring the backed up application", func() {
 			for _, namespace := range bkpNamespaces {
-				backupName := fmt.Sprintf("%s-%s", BackupNamePrefix, namespace)
-				CreateRestore("test-restore", backupName, namespaceMapping, destinationClusterName, orgID)
+				backupName = fmt.Sprintf("%s-%s", BackupNamePrefix, namespace)
+				restoreName = fmt.Sprintf("%s-%s", "test-restore", namespace)
+				CreateRestore(restoreName, backupName, namespaceMapping, destinationClusterName, orgID)
 			}
 		})
 	})
 	JustAfterEach(func() {
-		policyList := []string{intervalName, dailyName, weeklyName, monthlyName}
 		defer EndTorpedoTest()
-		for _, ruleName := range preRuleNameList {
-			err := Inst().Backup.DeleteRuleForBackup(orgID, ruleName)
-			err = fmt.Errorf("failed to delete pre rule %s with error %v", ruleName, err)
-			dash.VerifySafely(err, nil, "Deleting  backup rules")
+		policyList := []string{intervalName, dailyName, weeklyName, monthlyName}
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		backupDriver := Inst().Backup
+		backupUID, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
+		log.FailOnError(err, "Failed while trying to get backup UID for - %s", backupName)
+		DeleteBackup(backupName, backupUID, orgID, ctx)
+		DeleteRestore(restoreName, orgID)
+		if len(preRuleNameList) > 0 {
+			for _, ruleName := range preRuleNameList {
+				err := Inst().Backup.DeleteRuleForBackup(orgID, ruleName)
+				dash.VerifySafely(err, nil, "Deleting  backup pre rules")
+			}
 		}
-		for _, ruleName := range postRuleNameList {
-			err := Inst().Backup.DeleteRuleForBackup(orgID, ruleName)
-			err = fmt.Errorf("failed to delete post rule %s with error %v", ruleName, err)
-			dash.VerifySafely(err, nil, "Deleting  backup rules")
+		if len(postRuleNameList) > 0 {
+			for _, ruleName := range postRuleNameList {
+				err := Inst().Backup.DeleteRuleForBackup(orgID, ruleName)
+				dash.VerifySafely(err, nil, "Deleting  backup post rules")
+			}
 		}
-		err := Inst().Backup.DeleteBackupSchedulePolicy(orgID, policyList)
+		err = Inst().Backup.DeleteBackupSchedulePolicy(orgID, policyList)
 		dash.VerifySafely(err, nil, "Deleting backup schedule policies")
-		teardownStatus := TeardownForTestcase(contexts, providers, CloudCredUIDList)
-		dash.VerifyFatal(teardownStatus, true, "Testcase teardown status")
+		log.Info(" Deleting deployed applications")
+		for i := 0; i < len(contexts); i++ {
+			opts := make(map[string]bool)
+			opts[SkipClusterScopedObjects] = true
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
```

Comment:
see if you can use this one which validates and then deletes
 ValidateAndDestroy
if validation not required then fine

Reply:
done